Quotient/roommember.cpp

```diff
@@ -133,6 +134,19 @@ QUrl RoomMember::avatarUrl() const {
     return {};
 }
 
+int RoomMember::powerLevel() const
+{
+    if (_room == nullptr || _member == nullptr) {
+        return 0;
+    }
+
+    auto powerLevelEvent = _room->currentState().get<RoomPowerLevelsEvent>();
+    if (!powerLevelEvent) {
+        return 0;
```

Comment:
Unfortunately it's not that easy. In particular, absence of a power levels event means power level 100 for the room creator. I would suggest having a dedicated method in `Room` to calculate effective power level - I had it drafted but never got to pushing it. Here's the code, feel free to add it to the PR:
```cplusplus
int Room::effectivePowerLevel(const QString& userId) const
{
    if (!successorId().isEmpty())
        return false; // No one can upgrade a room that's already upgraded
    const auto& uId = !userId.isEmpty() ? userId : connection()->userId();
    const auto& curState = currentState();
    if (const auto* plEvent = curState.get<RoomPowerLevelsEvent>())
        return plEvent->powerLevelForUser(uId);
    if (const auto* createEvent = creation())
        return createEvent->senderId() == uId ? 100 : 0;
    return 0; // That's rather weird but may happen, according to rvdh
}
```

Reply:
Added along with a property to get the value for the local member which is I assume the point of returning the local member power level when ID is empty.